### PR TITLE
Cleanup of NGINX/Apache log file PRs #2155, #2156 & begin streamlining Munin playbook (WIP)

### PR DESCRIPTION
**Pull request by holta**
Includes & builds on @georgejhunt's PRs #2155, #2156 and my PR #2147.

### CODE REVIEW

**Comment by holta** [comment]
I'll work on this in the coming ~24h when I have more time &mdash; currently it fails on Ubuntu 19.10 (10.8.0.42) as follows:

> TASK [elgg : Restart Apache systemd service (apache2)] *************************
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "Unable to start service apache2: Job for apache2.service failed because the control process exited with error code.\nSee \"systemctl status apache2.service\" and \"journalctl -xe\" for details.\n"}

> root@box:~# systemctl status apache2
● apache2.service - The Apache HTTP Server
   Loaded: loaded (/lib/systemd/system/apache2.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Thu 2020-01-16 11:02:07 EST; 8min ago
     Docs: https://httpd.apache.org/docs/2.4/
  Process: 23616 ExecStart=/usr/sbin/apachectl start (code=exited, status=1/FAILURE)
>
> Jan 16 11:02:07 box.lan systemd[1]: Starting The Apache HTTP Server...
Jan 16 11:02:07 box.lan apachectl[23616]: (98)Address already in use: AH00072: make_sock: could not bind to address 0.0.0.0:80
Jan 16 11:02:07 box.lan apachectl[23616]: no listening sockets available, shutting down
Jan 16 11:02:07 box.lan apachectl[23616]: AH00015: Unable to open logs
Jan 16 11:02:07 box.lan apachectl[23616]: Action 'start' failed.
Jan 16 11:02:07 box.lan apachectl[23616]: The Apache error log may have more information.
Jan 16 11:02:07 box.lan systemd[1]: apache2.service: Control process exited, code=exited, status=1/FAILURE
Jan 16 11:02:07 box.lan systemd[1]: apache2.service: Failed with result 'exit-code'.
Jan 16 11:02:07 box.lan systemd[1]: Failed to start The Apache HTTP Server.

**Comment by tim-moody** [comment]
This PR is way too broad.

**Comment by tim-moody** [comment]
looks like apache and nginx are fighting over port 80. whichever runs first gets it.

**Comment by holta** [comment]
> looks like apache and nginx are fighting over port 80. whichever runs first gets it.

I strongly suspect it's a bug I introduced and will resolve when I get back to keyboard within ~24h.

**Comment by jvonau** [comment]
Is this PR/branch broken or master?

File: roles/nginx/tasks/main.yml [hunk 1]
```diff
@@ -16,9 +16,9 @@
     src: "{{ item.src}}"
     dest: "{{ item.dest }}"
   with_items:
-    - { src: 'server.conf', dest: '/etc/nginx/' }
-    - { src: 'nginx.conf', dest: '/etc/nginx/' }
-    - { src: 'ports.conf', dest: "/etc/{{ apache_service }}/" }
+    - { src: 'server.conf.j2', dest: '/etc/nginx/' }
+    - { src: 'nginx.conf.j2', dest: '/etc/nginx/' }
+    - { src: 'ports.conf.j2', dest: "/etc/{{ apache_service }}/" }
```
Comment (jvonau) [inline roles/nginx/tasks/main.yml]: need to use the file name when writing the file name.
dest ... /etc/nginx/server.conf
dest ... /etc/nginx/nginx.conf
dest ... /etc/{{ apache_service }}/ports.conf

File: roles/nginx/tasks/install.yml [hunk 2]
```diff
@@ -24,9 +24,9 @@
     src: "{{ item.src }}"
     dest: "{{ item.dest }}"
   with_items:
-    - { src: 'roles/nginx/templates/server.conf', dest: '/etc/nginx/' }
-    - { src: 'roles/nginx/templates/nginx.conf', dest: '/etc/nginx/' }
-    - { src: 'roles/nginx/templates/ports.conf', dest: '/etc/{{ apache_service }}/' }
+    - { src: 'roles/nginx/templates/server.conf.j2', dest: '/etc/nginx/' }
+    - { src: 'roles/nginx/templates/nginx.conf.j2', dest: '/etc/nginx/' }
+    - { src: 'roles/nginx/templates/ports.conf.j2', dest: '/etc/{{ apache_service }}/' }
```
Comment (jvonau) [inline roles/nginx/tasks/install.yml]: need to use the file name when writing the file name.
dest ... /etc/nginx/server.conf
dest ... /etc/nginx/nginx.conf
dest ... /etc/{{ apache_service }}/ports.conf

File: roles/captiveportal/tasks/main.yml [hunk 3]
```diff
@@ -53,34 +58,34 @@
...
+- name: Run iiab-divert-to-nginx to generate diversion lists for NGINX
   shell: /usr/sbin/iiab-divert-to-nginx
-     
-- name: Run script to generate nginx servers from checkurls input list
+
+- name: Run iiab-make-cp-servers.py to generate NGINX servers from checkurls input list (creates /etc/nginx/sites-available/capture.conf)
   command: /usr/sbin/iiab-make-cp-servers.py
   args:
-      creates: /etc/nginx/sites-available/capture.conf
+    creates: /etc/nginx/sites-available/capture.conf
```
Comment (jvonau) [inline roles/captiveportal/tasks/main.yml:85]: rest of the roles are using nginx_conf_dir in place of sites-available|enabled

File: roles/2-common/tasks/fl.yml [hunk 4]
```diff
@@ -29,10 +31,20 @@
...
 
-- name: Symlink from {{ doc_root }}/common/webfonts to {{ doc_root }}/common/fonts
+- name: File Layout - Symlink {{ doc_root }}/common/webfonts -> {{ doc_root }}/common/fonts
   file:
-    src: "{{ doc_root }}/common/fonts"
+    src: "{{ doc_root }}/common/fonts"    # /library/www/html
     path: "{{ doc_root }}/common/webfonts"
     state: link
+
+- name: File Layout - Create log file directories {{ apache_log_dir }} & {{ nginx_log_dir }}
```
Comment (jvonau) [inline roles/2-common/tasks/fl.yml:41]: should be handled automatically when the deb package is installed.

Comment (holta) [reply roles/2-common/tasks/fl.yml:41]: I'll comment out this stanza if you're sure?

Comment (holta) [reply roles/2-common/tasks/fl.yml:41]: Do we still care about setting these 2 dirs' owner/group as {{ apache_user }} (generally both 'www-data') and permission to '0770' ?

(Or do we let Apache / NGINX set their own defaults going forward?)

File: roles/captiveportal/tasks/main.yml (hunk 3 above)
Comment (holta) [reply roles/captiveportal/tasks/main.yml:85]: Actually the NGINX variable {{ nginx_config_dir }} = /etc/nginx/conf.d currently used by most/other roles — is a different path than the above.

But this does raise the question: are all these other roles using the recommended/best path?

Comment (jvonau) [reply roles/captiveportal/tasks/main.yml:85]: I did ask that question in #2042

Comment (holta) [reply roles/captiveportal/tasks/main.yml:85]: @m-anish can your NGINX expert/friend contribute an opinion as to whether IIAB's app sites (http://box/nodered, http://box/stream-fishing, etc!) should each place their .conf files (nodered-nginx.conf, stream-fishing-nginx.conf, etc) into `/etc/nginx/conf.d` or `/etc/nginx/sites-available` or whatever/wherever ?

**Comment by holta** [comment]
@georgejhunt or anybody: do you know why Captive Portal can fail to create `/opt/iiab/captiveportal/users.sqlite` below, unless you do a complete './iiab-install --reinstall' ?
https://github.com/holta/iiab/blob/cleanups/roles/captiveportal/templates/capture-wsgi.py.j2#L75

Which in turn can cause systemd service `uwsgi-captiveportal` to start incorrectly, with 'systemctl status uwsgi-captiveportal' messages like 'no python application found, check your startup logs for errors'.

SO BE IT, as at least it works as part of a complete './iiab-install --reinstall' !  But annoying that post-install things like './runrole captiveportal' and './runrole --reinstall captiveportal' cannot promise a valid (& rapid) Captive Portal _(hopefully something we can fix in future!)_

**Comment by holta** [comment]
Smoke-tested on Ubuntu 19.10 and Raspbian Lite.

(Testing has been extremely intensive, good enough for now: I'm merging it so we can focus on the most important bugs with Integration Testing across/with many other PR's taking this to the next level.)

Ref PRs: #2155 #2156 #2159 #2160

File: roles/2-common/tasks/fl.yml (hunk 4 above)
Comment (holta) [reply roles/2-common/tasks/fl.yml:41]: Indeed, it looks like NGINX's default ownership and permissions are somehow overwriting/overriding the actions of [fl.yml](https://github.com/iiab/iiab/blob/master/roles/2-common/tasks/fl.yml#L41-L50) here.

And consequently `/var/log/nginx` ownership/permissions seem to be is causing Captive Portal to fail (to create a log file there) at times, with more details here: #2168

**Pull request merged**